### ISSUE TITLE
compute advantages: wrong device

### ISSUE THREAD

**Issue by axelbr** [rlworkgroup/garage#2181]
In line 86 ([torch/_functions.py](https://github.com/rlworkgroup/garage/blob/f856f74eacb0d8f535eb36d9c84309efbc25c384/src/garage/torch/_functions.py#L86)), a runtime error is thrown when the rewards and baselines are located on another device. I suggest to either introduce an optional parameter `device` or querying it using `global_device()`.

Exact error message:
```
... line 86, in compute_advantages
advantages = F.conv2d(deltas, adv_filter, stride=1).reshape(rewards.shape)

RuntimeError: Input type (torch.cuda.FloatTensor) and weight type (torch.FloatTensor) should be the same
```

**Comment by avnishn** [comment]
Hi @axelbr thanks for checking out Garage and opening this issue!

while we could provide a device parameter, this would be the same as if a user made the following call to 
```python
compute_advantages(discount, gae_lambda, max_episode_length, baselines.to(device),
                       rewards.to(device))
```
I also am against using the global_device by default, as this can cause unexpected behaviors in terms of device ussage.

I think that the best way forward is to handle this the same way that torch does, which is to let users handle this on their own.

@krzentner, @irisliucy, what do you think?

**Comment by axelbr** [comment]
No, not entirely imo. What if i would want to compute the returns on the gpu? The adv_filters would still live on the cpu which leads to the described error.

**Comment by krzentner** [comment]
Ah, I see. Yeah, we should move the deltas and the advantage filter to the same device as the rewards before computing the convolution.

**Comment by axelbr** [comment]
Similar issue here: [gaussian_mlp_module.py](https://github.com/rlworkgroup/garage/blob/f856f74eacb0d8f535eb36d9c84309efbc25c384/src/garage/torch/modules/gaussian_mlp_module.py#L303).

After calling .to(), `self._init_std` will be on GPU, but the zero-vector is still on CPU.
